staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -473,8 +473,22 @@ func AddContainerVarFlags(cmd *cobra.Command, p *string, containerName string) {
 	cmd.Flags().StringVarP(p, "container", "c", containerName, "Container name. If omitted, use the kubectl.kubernetes.io/default-container annotation for selecting the container to be attached or the first container in the pod will be chosen")
 }
 
+var ServerSideAutoEnabledEnv = "KUBECTL_AUTO_SERVER_SIDE"
+
+func ServerSideAutoEnabled() bool {
+	return os.Getenv(ServerSideAutoEnabledEnv) == "true"
+}
+
```

Comment:
I'll update this to use the feature gate functionality introduce [here](https://github.com/kubernetes/kubernetes/pull/115979/files#diff-62da7750fb41c9d53934dd9706ba719f0c0efdf931b6b36eb46498505e4d3520R425-R435) once it's merged